src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp

```diff
@@ -0,0 +1,404 @@
+
+                        if (Creature* trigger = firstDoor->SummonTrigger(firstDoor->GetPositionX(), firstDoor->GetPositionY(), firstDoor->GetPositionZ(), 0, 500))
+                        {
+                            std::list<Creature*> defensorList;
+                            GetCreatureListWithEntryInGrid(defensorList, trigger, 65337, 20.0f);
+
+                            trigger->CastSpell(trigger, SPELL_EXPLOSION); // Explosion
+
+                            for (std::list<Creature*>::iterator itr = defensorList.begin(); itr != defensorList.end(); ++itr)
+                            {
+                                uint8 random = urand(0, 1);
+
+                                float posX = random ? 814.0f:  640.0f;
+                                float posY = random ? 2102.0f: 2112.0f;
+                                (*itr)->KnockbackFrom(posX, posY, 25.0f, 20.0f);
+                                (*itr)->DespawnOrUnsummon(1000);
+                            }
+                        }
+                    }
+                    break;
+                }
+                case DATA_BRASIER_CLICKED:
+                {
+                    if (dataStorage[type] == DONE)
+                        return;
+
+                    if (GetBossState(DATA_GADOK) != DONE)
+                        return;
+
+                    Map::PlayerList const &PlayerList = instance->GetPlayers();
+                    for (Map::PlayerList::const_iterator it = PlayerList.begin(); it != PlayerList.end(); ++it)
+                    {
+                        if (Player* player = it->GetSource())
+                        {
+                            player->SendCinematicStart(CINEMATIC_SETTING_SUN);
+                            player->SetInPhase(2, true, true);
+                            player->NearTeleportTo(1370.0f, 2283.6f, 402.328f, 2.70f);
+                        }
+                    }
+
+                    cinematicTimer = 100;
+                    dataStorage[type] = data;
+                    break;
+                }
+                default:
+                    if (type < MAX_DATA)
+                        dataStorage[type] = data;
+                    break;
+            }
+        }
+
+        uint32 GetData(uint32 type) const override
+        {
+            switch (type)
+            {
+                case DATA_OPEN_FIRST_DOOR:
+                default:
+                    if (type < MAX_DATA)
+                        return dataStorage[type];
+                    break;
+            }
+
+            return 0;
+        }
+
+        void SetData64(uint32 type, uint64 value)
+        {
+            switch (type)
+            {
+                case NPC_WEAK_SPOT:     raigonWeakGuid = GetObjectGuid(value);     break;
+                default:                                            break;
+            }
+        }
+
+        uint64 GetData64(uint32 type) const override 
+        {
+            switch (type)
+            {

```

Comment:
... if you are going to use this switch case with a different function or method, use TC codestyle:
```cpp
            {
                case NPC_KIPTILAK:
                    return kiptilakGuid.GetEntry();
                case NPC_GADOK:
                    return gadokGuid.GetEntry();
                case NPC_RIMOK:
                    return rimokGuid.GetEntry();
                case NPC_RAIGONN:
                    return raigonnGuid.GetEntry();
                case NPC_WEAK_SPOT:
                    return raigonWeakGuid.GetEntry();
                case DATA_RANDOM_BOMBARDER:
                    return Trinity::Containers::SelectRandomContainerElement(bombarderGuids).GetEntry();
                case DATA_RANDOM_BOMB_STALKER:
                    return Trinity::Containers::SelectRandomContainerElement(bombStalkerGuids).GetEntry();
            }
```